easydigitaldownloads/easy-digital-downloads
# Change client side validation when a discount is applied

Fixes #7487

Proposed Changes:
1. When a discount is applied, only fields other than `.card-address-2` should be marked as `required`.

@spencerfinnell this is different than what you originally proposed as the `input` was not an object. Making the logic more verbose felt more readable to me.

To test:
1. Make sure taxes are disabled.
2. This may happen with any gateway, but it's easiest to test with Stripe, so enable that.
3. Do not collect address information on checkout.
4. Add something to your cart.
5. Make sure that Stripe/Credit Card is selected as the payment method, and have at least one saved card (this forces the add new card form to be hidden).
6. Apply a discount to the cart.
7. Complete your purchase. In `master`, you will not be able to complete this step due to a JS error, but in this branch, you should be able to.
